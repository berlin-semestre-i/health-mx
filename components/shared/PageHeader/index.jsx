import React from 'react'
import Moment from 'react-moment'
import styled from 'styled-components'
import { Grid, Header } from 'semantic-ui-react'
import media from 'styled-media-query'

const PageHeader = ({title, subtitle}) => (
  <React.Fragment>
    <Grid.Row>
      <Grid.Column mobile={16} computer={12}>
        <CustomHeader as="h3">{title}</CustomHeader>
        <CustomHeader as="h1">{subtitle}</CustomHeader>
      </Grid.Column>
      <Grid.Column mobile={16} computer={4}>
        <DateContainer>
          <Date className="ui header" element="h3" locale="es" format="LL" />
          <Date element="h3" locale="es" format="hh:mm a" interval={15000} />
        </DateContainer>
      </Grid.Column>
    </Grid.Row>
  </React.Fragment>
)

export default PageHeader

const DateContainer = styled.div`

  ${media.greaterThan('1024px')`
    position: absolute;
    bottom: 0;
    right: 1rem;
    text-align: right;
  `}
  
`
const CustomHeader = styled(Header)`
  &&& {
    margin: 0;
  }
`

const Date = styled(Moment)`
  &&& {
    letter-spacing: inherit;
    text-transform: inherit;
    font-size: 17px;
    margin: 0;
  }
`