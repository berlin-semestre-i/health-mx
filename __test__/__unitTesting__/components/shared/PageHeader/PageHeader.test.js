import React from 'react'
import { shallow, mount } from 'enzyme'
import PageHeader from '../../../../../components/shared/PageHeader'
import wait from 'waait'

Date.now = jest.fn(() => 1487076719000)

describe('Header', () => {
  it('renders', () => {
    const component = shallow(<PageHeader title="" subtitle="" />)
    expect(component).toMatchSnapshot()
  })
})