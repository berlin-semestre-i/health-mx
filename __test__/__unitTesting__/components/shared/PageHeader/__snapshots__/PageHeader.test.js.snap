// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<Fragment>
  <GridRow>
    <GridColumn
      computer={12}
      mobile={16}
    >
      <PageHeader__CustomHeader
        as="h3"
      />
      <PageHeader__CustomHeader
        as="h1"
      />
    </GridColumn>
    <GridColumn
      computer={4}
      mobile={16}
    >
      <PageHeader__DateContainer>
        <PageHeader__Date
          className="ui header"
          element="h3"
          format="LL"
          locale="es"
        />
        <PageHeader__Date
          element="h3"
          format="hh:mm a"
          interval={15000}
          locale="es"
        />
      </PageHeader__DateContainer>
    </GridColumn>
  </GridRow>
</Fragment>
`;
